### DISCUSSION TITLE
Eclipse Prediction

### DISCUSSION THREAD

**Discussion by tgflynn** [Stellarium/stellarium#3637] [General]
Hi. 

I'd like to be able to make accurate solar eclipse predictions (ie. start and end of partial and total eclipse) for arbitrary lat/long coordinates and I'm thinking that stellarium would probably be helpful for this rather than doing all the calculations from scratch.  

Does stellarium have any existing feature or script for doing this ?

If not could anyone point me in the right general direction for how to implement this with a stellarium script, or plugin ?

Thanks.

**Comment by 10110111** [comment]
User Guide, chapter 4.6.8.

**Comment by tgflynn** [comment]
While the Eclipse tab of the astronomical calculations window provides a lot of information, it's not exactly what I'm looking for.  I want to generate more precise eclipse data for coordinates within a smaller geographical area, unlike the kml file which covers the entire earth.

Is there a way to access the calculated eclipse data through the scripting interface ?  Or more generally can you access data about celestial objects through scripting ?  The script examples I have found only seem to show automation of the UI but not extraction of data.

**Reply by alex-w** [reply]
The second tab - Local solar eclipses - does exactly what you want

**Comment by tgflynn** [comment]
But that only shows the information for one specific coordinate.  I want to be able to generate that data for a number of points over a certain geographical area.

**Comment by tgflynn** [comment]
So it looks like all I needed was core.getObjectInfo["Sun"].  That returns an object that contains the current eclipse magnitude along with other data.  With that and the script documentation on how to set times and locations I think should be able to generate the data I want.

**Reply by gzotti** [reply]
Yes, if you want to test many locations and find some optimum, or to get a list/ephemeris of conditions for 200 different locations to present in a different tabular format, such scripting approach is good.

**Reply by 1badapple** [reply]
Too bad Stellarium doesn't have a simple, point and click, option to generate what you need. Imagine if ANY planet/star eclipses trajectory could be easily plotted and identified. Sounds like a must-have option, in future app versions

**Reply by gzotti** [reply]
https://github.com/Stellarium/stellarium/wiki/FAQ#why-dont-you-implement
Seriously, there are enough dedicated websites for eclipse data lists. Stellarium is about visualisation, not data lists. We are open for your contributions in working code, though.
